packages/next/pages/_document.tsx

```diff
@@ -724,5 +730,5 @@ function getPageFile(page: string, buildId?: string) {
     return buildId ? `/index.${buildId}.js` : '/index.js'
   }
 
-  return buildId ? `${page}.${buildId}.js` : `${page}.js`
+  return encodeURI(buildId ? `${page}.${buildId}.js` : `${page}.js`)
```

Comment:
This shouldn't happen here, but in individual script tags itself IMO. It'd be too easy to double-encode by accident.